api/v1beta1/flinkcluster_types.go

```diff
@@ -347,6 +348,9 @@ type JobSpec struct {
 	// Allow non-restored state, default: false.
 	AllowNonRestoredState *bool `json:"allowNonRestoredState,omitempty"`
 
+	// Should take savepoint before upgrading the job, default: false.
+	ShouldTakeSavepointOnUpgrade *bool `json:"shouldTakeSavepointOnUpgrade,omitempty"`
```

Comment:
nit: s/ShouldTakeSavepointOnUpgrade/TakeSavepointOnUpdate.

Reply:
@shashken It would be nice to consider setting a default value for `takeSavepointOnUpgrade` field. Otherwise it could lead to a nil error like #408. You can easily set it with a marker like `kubebuilder:default`. 

And considering the naming consistency, the field name `TakeSavepointOnUpdate` looks better to me.

Reply:
@shashken And the default value is `false` in the comment. Wouldn't it be better to set it to `true`? It seems common to restore the job from the latest savepoint.

Reply:
@shashken @functicons And the default value is `false` in the comment. Wouldn't it be better to set it to `true`? It seems common to restore the job from the latest savepoint.